src/vision/status_quo/alan_hates_writing_a_stream.md

```diff
@@ -0,0 +1,194 @@
+# 😱 Status quo stories: Alan hates writing a `Stream`
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+[Alan][] is used to writing web server applications using async sockets, but wants to try Rust to get that signature vroom vroom.
+
+After a couple weeks learning Rust basics, Alan quickly understands `async` and `await`, and therefore has several routes built for his application that await a few things and then construct an HTTP response and send a buffered body. To build the buffered response bodies, Alan was reading a file, and then appending a signature, and putting that all into a single buffer of bytes.
+
+Eventually, Alan realizes that some responses have enormous bodies, and would like to stream them instead of buffering them fully in memory. He's *used* the `Stream` trait before. Using it was very natural, and followed a similar pattern to regular `async`/`await`:
+
+```rust
+while let Some(chunk) = body.next().await? {
+    file.write_all(&chunk).await?;
+}
+```
+
+However, _implementing_ `Stream` turns out to be rather different. While he quickly learned the simple way to turn a `File` into a `Stream` with `StreamReader`, the chaining part was much harder.
```

Comment:
Question: how did Alan learn about `StreamReader`? Were the docs for chaining streams not mentioned in those docs? It would be interesting to figure out where the breakdown in docs was exactly.

Reply:
Also, don't you mean `ReaderStream`?